plugins/linear-genome-view/src/LinearGenomeView/components/OverviewScaleBar.tsx

```diff
@@ -150,83 +200,120 @@ type LGV = Instance<LinearGenomeViewStateModel>
 
 const ScaleBar = observer(({ model, scale }: { model: LGV; scale: number }) => {
   const classes = useStyles()
-
-  const { displayedParentRegions, dynamicBlocks: visibleRegions } = model
+  const { displayedRegions, dynamicBlocks: visibleRegions } = model
   const { assemblyManager } = getSession(model)
-
   const gridPitch = chooseGridPitch(scale, 120, 15)
 
   return (
     <div className={classes.scaleBar}>
-      {displayedParentRegions.map((seq, idx) => {
+      {/* this is the entire scale bar */}
+      {displayedRegions.map((seq, idx) => {
         const assembly = assemblyManager.get(seq.assemblyName)
         let refNameColor: string | undefined
         if (assembly) {
           refNameColor = assembly.getRefNameColor(seq.refName)
         }
         const regionLength = seq.end - seq.start
+        // boolean if displayed region length is smaller than its parent's
+        const incompleteRegion =
+          seq.parentEnd - seq.parentStart > seq.end - seq.start
+
         const numLabels = Math.floor(regionLength / gridPitch.majorPitch)
         const labels = []
         for (let index = 0; index < numLabels; index++) {
-          labels.push((index + 1) * gridPitch.majorPitch)
+          const offsetLabel = (index + 1) * gridPitch.majorPitch
+          const label = seq.reversed
+            ? seq.end - offsetLabel
+            : offsetLabel + seq.start
+          labels.push(label)
         }
-
         return (
-          // each whole sequence
           <Paper
-            key={seq.refName}
+            key={`${JSON.stringify(seq)}-${idx.toLocaleString('en-US')}`}
+            variant="outlined"
+            className={clsx(
+              classes.scaleBarContig,
+              seq.reversed
+                ? classes.scaleBarContigReverse
+                : classes.scaleBarContigForward,
+            )}
             style={{
               minWidth: regionLength / scale,
               marginRight:
-                idx === displayedParentRegions.length - 1
+                idx === displayedRegions.length - 1
                   ? undefined
                   : wholeSeqSpacer,
               borderColor: refNameColor,
             }}
-            className={classes.scaleBarContig}
-            variant="outlined"
           >
+            {incompleteRegion && (
+              <div
+                className={classes.scaleBarRegionIncompleteLeft}
+                style={{
+                  backgroundImage: `linear-gradient(-225deg, ${refNameColor} 3px, transparent 1px),
+                linear-gradient(45deg, ${refNameColor} 3px, transparent 1px)`,
```

Comment:
can we move these backgroundImages into a useStyles class

Reply:
is the answer no because of refNameColor?

Reply:
Yep thats the only reason why that style is there.